src/components/rate/index.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { HTMLAttributes } from 'react';
+
+import { CSS } from '../../system';
+
+import { forwardRef } from '../../utils/forwardRef';
+
+import { Stack } from '../stack';
+import { VisuallyHidden } from '../visually-hidden';
+
+import { StarIcon } from './icon';
+
+import * as S from './styles';
+
+export type RateProps = {
+  /**
+   * Total number of selected stars
+   */
+  value: number;
+  /**
+   * CSS properties
+   */
+  css?: CSS;
+} & HTMLAttributes<HTMLDivElement>;
+
+/**
+ * Rate component
+ *
+ * @description Ratings provide insight regarding others' opinions and experiences.
+ */
+export const Rate = forwardRef<RateProps, 'div'>((props) => {
+  const { value, ...rest } = props;
+
+  if (value < 0 || value > 5) {
+    throw new Error('Rate value must be between 0 and 5');
+  }
+
+  const totalStars = [...Array(5)];
+
+  // pluralize the label
+  const pluralStar = value > 1 ? 'stars' : 'star';
+
+  // message to be read by screen readers
+  const a11yMessage = `${value} out of 5 ${pluralStar}`;
```

Comment:
You don't need to pluralize the sentence. The correct will be: "1 out of 5 stars"

Reply:
yep, you right. I will fix that